compiler/rustc_serialize/src/opaque.rs

```diff
@@ -346,7 +341,7 @@ impl FileEncoder {
             // room to write the input to the buffer.
             unsafe {
                 let src = buf.as_ptr();
-                let dst = MaybeUninit::slice_as_mut_ptr(&mut self.buf).add(buffered);
+                let dst = self.buf[buffered].as_mut_ptr();
```

Comment:
This adds a runtime bounds check where there was none before. Same for all the other places where `add` became `[...]`.

Reply:
Correct. But the unsafety is centered around `copy_nonoverlapping` and presumably llvm can optimize out the bounds checking since it's done right above?

Happy to remove the bounds checking if that's what reviewers want, but either way `get_unchecked` is the method designed to express the desire to avoid bounds checking, not pointer addition IMO.

Reply:
I don't know if the bounds checks here are a problem, I only know I took care to not introduce any new bounds checks when this code was originally ported to `MaybeUninit`.

Reply:
I went back through and removed all the bounds checks that wouldn't be optimized out by the compiler. Anything remaining should be trivially removable.